step-functions/nino_issue_credential.asl.json

```diff
@@ -1,429 +1,456 @@
+						},
+						"Add Type to VC": {
+							"Type": "Pass",
+							"Next": "Did the user fail or pass the check?",
+							"Parameters": {
+								"type": [
+									"VerifiableCredential",
+									"IdentityCheckCredential"
+								],
+								"@context": [
+									"https://www.w3.org/2018/credentials/v1",
+									"https://vocab.london.cloudapps.digital/contexts/identity-v1.jsonld"
+								]
+							},
+							"ResultPath": "$.vctype"
+						},
+						"Did the user fail or pass the check?": {
+							"Type": "Choice",
+							"Choices": [
+								{
+									"Or": [
+										{
+											"Variable": "$.check-attempts-exist.Count",
+											"NumericGreaterThanEquals": 2
+										}
+									],
+									"Next": "Fetch CI"
+								}
+							],
+							"Default": "create VC evidence of a pass"
+						},
+						"create VC evidence of a pass": {
+							"Type": "Pass",
+							"Parameters": {
+								"type": "IdentityCheck",
+								"txn.$": "States.UUID()",
+								"strengthScore": 2,
+								"validityScore": 2,
+								"checkDetails": [
+									{
+										"checkMethod": "data"
+									}
+								]
+							},
+							"ResultPath": "$.evidence",
+							"Next": "create Audit evidence of a pass"
+						},
+						"create Audit evidence of a pass": {
+							"Type": "Pass",
+							"Parameters": {
+								"evidence.$": "$.evidence"
+							},
+							"ResultPath": "$.audit",
+							"End": true
+						},
+						"Fetch CI": {
+							"Type": "Task",
+							"Resource": "arn:aws:states:::lambda:invoke",
+							"Parameters": {
+								"FunctionName": "${CiMappingFunctionArn}",
+								"Payload": {
+									"ci_mapping.$": "States.StringSplit($.parameters.contraindicationMappings,'||')",
+									"hmrc_errors.$": "$.check-attempts-exist.Items[*].text.S",
+                  "ci_reason_mapping.$": "States.StringToJson($.parameters.contraindicatorReasonMappings)"
+								}
+							},
+							"Next": "create VC evidence of a failure",
+							"ResultPath": "$.contraIndicators"
+						},
+						"create VC evidence of a failure": {
+							"Type": "Pass",
+							"Parameters": {
+								"type": "IdentityCheck",
+								"txn.$": "States.UUID()",
+								"strengthScore": 2,
+								"validityScore": 0,
+								"failedCheckDetails": [
+									{
+										"checkMethod": "data"
+									}
+								],
+								"ci.$": "States.ArrayUnique($.contraIndicators.Payload[*].ci)"
+							},
+							"Next": "create Audit evidence of a failure",
+							"ResultPath": "$.evidence"
+						},
+						"create Audit evidence of a failure": {
+							"Type": "Pass",
+							"Parameters": {
+								"evidence": {
+									"type": "IdentityCheck",
+									"txn.$": "States.UUID()",
+									"strengthScore": 2,
+									"validityScore": 0,
+									"failedCheckDetails": [
+										{
+											"checkMethod": "data"
+										}
+									],
+									"ci.$": "States.ArrayUnique($.contraIndicators.Payload[*].ci)",
```

Comment:
If we have `ArrayUnique` here, do we definitely need a `deduplicateContraIndicators` function too?